kafka/conn.py

```diff
@@ -834,50 +834,42 @@ def _recv(self):
         """Take all available bytes from socket, return list of any responses from parser"""
         recvd = []
         err = None
+
         with self._lock:
             if not self._can_send_recv():
                 log.warning('%s cannot recv: socket not connected', self)
                 return ()
 
-            while len(recvd) < self.config['sock_chunk_buffer_count']:
-                try:
-                    data = self._sock.recv(self.config['sock_chunk_bytes'])
-                    # We expect socket.recv to raise an exception if there are no
-                    # bytes available to read from the socket in non-blocking mode.
-                    # but if the socket is disconnected, we will get empty data
-                    # without an exception raised
-                    if not data:
-                        log.error('%s: socket disconnected', self)
-                        err = Errors.KafkaConnectionError('socket disconnected')
-                        break
-                    else:
+            try:
+                # BlockingIOError is a PY2 catchall that should not be caught
```

Comment:
BlockingIOError was actually not getting reraised before, so I'm unsure if this is actually the correct intended behavior